File: app/src/main/java/com/omvp/app/interceptor/authPhone/AuthPhoneInterceptorCallback.java
package com.omvp.app.interceptor.authPhone;

import com.google.firebase.auth.FirebaseUser;
import com.raxdenstudios.square.interceptor.InterceptorCallback;

public interface AuthPhoneInterceptorCallback extends InterceptorCallback {

    void authPhoneUserRetrieved(FirebaseUser firebaseUser);

    void authPhoneCodeRetrieved(String code);

    void authPhoneStateChanged(AuthPhoneInterceptor.AuthPhoneState authPhoneState);

    void authPhoneError(AuthPhoneInterceptor.AuthPhoneError authPhoneError);

}

File: app/src/main/java/com/omvp/app/interceptor/authPhone/AuthPhoneInterceptorCallback.kt
package com.omvp.app.interceptor.authPhone

import com.google.firebase.auth.FirebaseUser
import com.raxdenstudios.square.interceptor.InterceptorCallback

interface AuthPhoneInterceptorCallback : InterceptorCallback {

    fun authPhoneUserRetrieved(firebaseUser: FirebaseUser)

    fun authPhoneCodeRetrieved(code: String)

    fun authPhoneStateChanged(authPhoneState: AuthPhoneInterceptor.AuthPhoneState)

    fun authPhoneError(authPhoneError: AuthPhoneInterceptor.AuthPhoneError)

}
